sqlglot/optimizer/qualify_columns.py

```diff
@@ -220,11 +220,18 @@ def _update_source_columns(source_name: str) -> None:
             if not column.table and column.name in column_tables:
                 tables = column_tables[column.name]
                 coalesce_args = [exp.column(column.name, table=table) for table in tables]
-                replacement = exp.func("coalesce", *coalesce_args)
+                replacement: exp.Expression = exp.func("coalesce", *coalesce_args)
 
-                # Ensure selects keep their output name
                 if isinstance(column.parent, exp.Select):
+                    # Ensure the USING column keeps it's output name if it's projected
                     replacement = alias(replacement, alias=column.name, copy=False)
+                elif not isinstance(column.parent, exp.PropertyEQ) and column.find_ancestor(
+                    exp.Struct
+                ):
```

Comment:
Can't we just do `if isinstance(column.parent, exp.Struct)`?

Reply:
Yeah I think that simplifies the condition, also doesn't make a difference for nested structs